processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -246,24 +261,25 @@ public void run() {
             } catch (RuntimeException e) {
                 log.error("Offset commit failed before closing consumer", e);
             }
-            contexts.close();
-
             processors.destroySingletonScope(scope.subscriptionId());
```

Comment:
H-m, better to move to `cleanUp` ?
I think there's a race like below:

Say user is enabled DC timeout and retry feature.

1. `processors.destroySingletonScope(scope.subscriptionId());`
    * retry processors accepts no more tasks from now on
2. timeout callback (executed on OffsetStateReaper thread) is taking long time (for user-supplied logic)
3. after a while, timeout callback decides the task to retry, then executed `ctx.retry()`
4. this spits garbage ERROR log because producer is already closed

Reply:
We found that this race also could happen for user-provided processors, because we can't ensure no more retry tasks are queued after destroying retry processor.
e.g:
1. processors.destroySingletonScope(scope.subscriptionId());
    * retry processors accepts no more tasks from now on
2. some tasks are still executed asynchronously on some thread
3. eventually `ctx.retry()` is called. but retry processor is already closed

So, current PR's execution order (destroySingletonScope => contexts.close()) doesn't change semantics much at least.

So current impl looks fine.